src/overcooked_ai_py/agents/benchmarking.py

```diff
@@ -19,36 +20,76 @@ class AgentEvaluator(object):
     pickleable. We should think about possible improvements/what makes most sense to do here.
     """
 
-
-    def __init__(self, mdp_params={}, env_params={}, mdp_fn=None, force_compute=False, mlam_params=NO_COUNTERS_PARAMS, debug=False):
+    def __init__(self, env_params, mdp_fn, force_compute=False, mlam_params=NO_COUNTERS_PARAMS, debug=False):
         """
-        mdp_params (dict): params for creation of an OvercookedGridworld instance through the `from_layout_name` method
         env_params (dict): params for creation of an OvercookedEnv
         mdp_fn (callable function): a function that can be used to create mdp
         force_compute (bool): whether should re-compute MediumLevelActionManager although matching file is found
         mlam_params (dict): the parameters for mlam, the MediumLevelActionManager
         debug (bool): whether to display debugging information on init
         """
-        assert type(mdp_params) is dict, "mdp_params must be a dictionary"
-        assert (mdp_params != {} and env_params != {}) or mdp_fn != None, "either evaluate from params or fn"
+        assert callable(mdp_fn), "mdp generating function must be a callable function"
         env_params["mlam_params"] = mlam_params
-        if mdp_fn is None:
+        self.mdp_fn = mdp_fn
+        self.env = OvercookedEnv(self.mdp_fn, **env_params)
+        self.force_compute = force_compute
+
+    @staticmethod
+    def from_mdp_params(mdp_params, env_params, outer_shape=None, mdp_params_schedule_fn=None, force_compute=False,
+                        mlam_params=NO_COUNTERS_PARAMS, debug=False):
+        """
+        mdp_params (dict): params for creation of an OvercookedGridworld instance through the `from_layout_name` method
+        outer_shape: the outer shape of environment
+        mdp_params_schedule_fn: the schedule for varying mdp params
+        Information for the rest of params please refer to the __init__ method above
+        """
+        # special case: generate from layout name
+        if type(mdp_params) is dict and "layout_name" in mdp_params:
             mdp = OvercookedGridworld.from_layout_name(**mdp_params)
-            self.mdp_fn = lambda: mdp
-            self.env = OvercookedEnv.from_mdp(mdp, **env_params)
-        else:
+            mdp_fn = lambda: mdp
+        else: # general case: variable mdp
+            assert outer_shape is not None, "outer_shape needs to be defined for variable mdp"
+            mdp_fn_naive = LayoutGenerator.mdp_gen_fn_from_dict(mdp_params, outer_shape, mdp_params_schedule_fn)
             # infinite mdp
-            if 'num_mdp' not in env_params or env_params['num_mdp'] == -1:
-                self.mdp_fn = mdp_fn
-                self.env = OvercookedEnv(self.mdp_fn, **env_params)
-            else:
+            if 'num_mdp' not in env_params or np.isinf(env_params['num_mdp']):
```

Comment:
We should probably operate under the assumption that num_mdp is always in env_params – it's more in line with offensive programming practices (that I feel like are better in research-code, as we want to catch errors or abstraction violations as soon as possible, and don't care about having code that errors when something is wrong – errors are not costly, as they don't impact any users or anything).